lib/cereal/patterns/singletaskworkflow.go

```diff
@@ -0,0 +1,54 @@
+package patterns
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/chef/automate/lib/cereal"
+	"github.com/sirupsen/logrus"
+)
+
+type SingleTaskWorkflowExecutor struct {
+	taskName    string
+	allowCancel bool
+}
+
+func NewSingleTaskWorkflowExecutor(taskName string, allowCancel bool) *SingleTaskWorkflowExecutor {
+	return &SingleTaskWorkflowExecutor{
+		taskName:    taskName,
+		allowCancel: allowCancel,
+	}
+}
+
+func (s *SingleTaskWorkflowExecutor) OnStart(w cereal.WorkflowInstance, ev cereal.StartEvent) cereal.Decision {
+	var params json.RawMessage
+	err := w.GetParameters(&params)
+	if err != nil {
+		logrus.WithError(err).Error("failed to get parameters")
+		w.Complete()
```

Comment:
I think you mean to return this.  We probably also want to fail the workflow:
```suggestion
		return w.Fail(err)
```